### PR TITLE
Change the provider implementation of X942kdf to use wpacket to do der encoding of sharedInfo

### PR DESCRIPTION
**Pull request by slontis**
Added der_writer functions for writing octet string primitives.

As @levitte has already found - there was a bug in the writer (it was writing 0x80 instead of 0xA0) for constructed blocks.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
This PR will become a dependency of a new PR related to supporting dhx in the providers.

File: crypto/der_writer.c [hunk 1]
```diff
@@ -62,6 +64,30 @@ int DER_w_boolean(WPACKET *pkt, int tag, int b)
...
+{
+    return int_start_context(pkt, tag)
+        && WPACKET_start_sub_packet(pkt)
+        && WPACKET_memcpy(pkt, data, data_n)
+        && WPACKET_close(pkt)
+        && WPACKET_put_bytes_u8(pkt, DER_P_OCTET_STRING)
+        && int_end_context(pkt, tag);
+}
+
+int DER_w_octet_string_uint32(WPACKET *pkt, int tag, size_t value)
```
Comment (mattcaswell) [inline crypto/der_writer.c:78]: Why not use `uint32_t` instead of `size_t` for `value`?

Comment (slontis) [reply crypto/der_writer.c:78]: yep - good point.

**Comment by levitte** [comment]
> As @levitte has already found - there was a bug in the writer (it was writing 0x80 instead of 0xA0) for constructed blocks.

Context-specific tags, specifically...  they aren't the easiest to deal with, what with different encodings if the item is declared explicit or implicit, and a bunch of other conditions (X.690 p 11 has a few examples).  The correction I've made has them explicit instead of implicit, and I hope that will be good enough for now.

**Comment by slontis** [comment]
Rebased - the only difference was in the der_writer - the change was the same. So it should not need reapproval.

**Comment by slontis** [comment]
Thanks. Merged to master.

**Pull request closed without merging**